File: lib/commands/builtin/echo.ts
import type { CommandSpec } from "../types"

export const echoCommand: CommandSpec = {
  name: "echo",
  summary: "print arguments",
  usage: ["echo [text...]"],
  man: ["NAME", "  echo - print arguments", "", "SYNOPSIS", "  echo [text...]"],
  async execute(_ctx, args) {
    return [args.slice(1).join(" ") || "(empty)"]
  }
}

File: wasm/bmxt-core/src/cmd/echo.rs
use crate::meta::Cmd;
use crate::model::DispatchJson;

pub const CMD: Cmd = Cmd {
    name: "echo",
    aliases: &[],
    usage_primary: "echo [text...]",
    man: &["NAME", "  echo - print arguments", "", "SYNOPSIS", "  echo [text...]"],
};

pub fn run(args: &[String]) -> DispatchJson {
    let joined = args.iter().skip(1).cloned().collect::<Vec<_>>().join(" ");
    DispatchJson::lines(vec![format_echo(&joined)])
}

fn format_echo(joined: &str) -> String {
    if joined.is_empty() {
        "(empty)".to_string()
    } else {
        joined.to_string()
    }
}
